cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1294,4 +1389,62 @@ private boolean canHaveInstances(ProgramType programType) {
     }
     return programs;
   }
+
+  private void updateLogLevels(HttpRequest request, HttpResponder responder, String namespace, String appName,
+                               String appVersion, String type, String programName, @Nullable String component)
+    throws Exception {
+    ProgramType programType = getProgramType(type);
+    if (programType == null) {
+      throw new MethodNotAllowedException(request.getMethod(), request.getUri());
+    }
+    try {
+      // we are decoding the body to Map<String, String> instead of Map<String, LogEntry.Level> here since Gson will
+      // serialize invalid enum values to null, which is allowed for log level, instead of throw an Exception.
+      lifecycleService.updateProgramLogLevels(
+        new ApplicationId(namespace, appName, appVersion).program(programType, programName),
+        transformLogLevelsMap(decodeArguments(request)), component);
+      responder.sendStatus(HttpResponseStatus.OK);
+    } catch (JsonSyntaxException e) {
+      throw new BadRequestException("Invalid JSON in body");
+    } catch (SecurityException e) {
+      throw new UnauthorizedException("Unauthorized to update the log levels");
+    }
+  }
+
+  private void resetLogLevels(HttpRequest request, HttpResponder responder, String namespace, String appName,
+                              String appVersion, String type, String programName, @Nullable String component)
+    throws Exception {
+    ProgramType programType = getProgramType(type);
+    if (programType == null) {
+      throw new MethodNotAllowedException(request.getMethod(), request.getUri());
+    }
+    try {
+      Map<String, List<String>> resetRequest = parseBody(request, STRING_STRINGARRAY_MAP_TYPE);
```

Comment:
What's the format of the request body? From the javadoc above, it sounds like it should be just a list of string?